packages/permissionless/accounts/kernel/signerToEcdsaKernelSmartAccount.ts

```diff
@@ -73,52 +89,175 @@ const createAccountAbi = [
     }
 ] as const
 
+export type KernelVersion = "0.2.2" | "0.3.0-beta"
+
 /**
- * Default addresses for kernel smart account
+ * Default addresses map for different kernel smart account versions
  */
-const KERNEL_ADDRESSES: {
-    ECDSA_VALIDATOR: Address
-    ACCOUNT_V2_2_LOGIC: Address
-    FACTORY_ADDRESS: Address
+export const KERNEL_VERSION_TO_ADDRESSES_MAP: {
+    [key in KernelVersion]: {
+        ECDSA_VALIDATOR: Address
+        ACCOUNT_LOGIC: Address
+        FACTORY_ADDRESS: Address
+        META_FACTORY_ADDRESS?: Address
+    }
 } = {
-    ECDSA_VALIDATOR: "0xd9AB5096a832b9ce79914329DAEE236f8Eea0390",
-    ACCOUNT_V2_2_LOGIC: "0x0DA6a956B9488eD4dd761E59f52FDc6c8068E6B5",
-    FACTORY_ADDRESS: "0x5de4839a76cf55d0c90e2061ef4386d962E15ae3"
+    "0.2.2": {
+        ECDSA_VALIDATOR: "0xd9AB5096a832b9ce79914329DAEE236f8Eea0390",
+        ACCOUNT_LOGIC: "0x0DA6a956B9488eD4dd761E59f52FDc6c8068E6B5",
+        FACTORY_ADDRESS: "0x5de4839a76cf55d0c90e2061ef4386d962E15ae3"
+    },
+    "0.3.0-beta": {
+        ECDSA_VALIDATOR: "0x8104e3Ad430EA6d354d013A6789fDFc71E671c43",
+        ACCOUNT_LOGIC: "0x94F097E1ebEB4ecA3AAE54cabb08905B239A7D27",
+        FACTORY_ADDRESS: "0x6723b44Abeec4E71eBE3232BD5B455805baDD22f",
+        META_FACTORY_ADDRESS: "0xd703aaE79538628d27099B8c4f621bE4CCd142d5"
+    }
+}
+
+/**
+ * Get supported Kernel Smart Account version based on entryPoint
+ * @param entryPoint
+ */
+const getKernelVersion = (entryPoint: EntryPoint): KernelVersion => {
```

Comment:
we can use entryPoint constant, also is the new version of kernel not usable with older entryPoint?

Reply:
Will use the constant. Missed that.